src/r1cs_nark_pcd/help_circuit.rs

```diff
@@ -0,0 +1,273 @@
+    pub(crate) fn compute_hash_var(
+        cs: ConstraintSystemRef<HelpField<E>>,
+        main_avk_var: &r1cs_nark_as::constraints::VerifierKeyVar<HelpField<E>>,
+        main_accumulator_instance_var: &AccumulatorInstanceVar<MainAffine<E>, PC::MainCurveVar>,
+    ) -> Result<FpVar<HelpField<E>>, SynthesisError> {
+        let mut sponge = PC::HelpSpongeVar::new(cs);
+        absorb_gadget!(&mut sponge, main_avk_var, main_accumulator_instance_var);
+        Ok(sponge.squeeze_field_elements(1)?.pop().unwrap())
+    }
+
+    /// Computes the public input of the help circuit.
+    pub(crate) fn compute_public_input(
+        main_avk: &r1cs_nark_as::VerifierKey,
+        main_accumulator_instance: &AccumulatorInstance<MainAffine<E>>,
+    ) -> Result<Vec<HelpField<E>>, SynthesisError> {
+        let cs = ConstraintSystem::<HelpField<E>>::new_ref();
+
+        let _hash = FpVar::new_input(cs.clone(), || {
+            Ok(Self::compute_hash(main_avk, main_accumulator_instance))
+        })?;
+
+        let r1cs_input = cs.borrow().unwrap().instance_assignment.clone();
+        Ok(r1cs_input)
+    }
+}
+
+impl<E, PC, P> ConstraintSynthesizer<HelpField<E>> for HelpCircuit<E, PC, P>
+where
+    E: CurveCycle,
+    MainField<E>: PrimeField + Absorbable<MainField<E>>,
+    HelpField<E>: PrimeField + Absorbable<HelpField<E>>,
+    MainAffine<E>: Absorbable<HelpField<E>>,
+    HelpAffine<E>: Absorbable<MainField<E>>,
+    PC: R1CSNarkPCDConfig<E>,
+    P: PCDPredicate<MainField<E>>,
+{
+    fn generate_constraints(
+        self,
+        cs: ConstraintSystemRef<HelpField<E>>,
+    ) -> Result<(), SynthesisError> {
+        let HelpCircuit {
+            main_avk,
+            main_accumulation_input_instances: main_input_instances,
+            main_old_accumulator_instances,
+            main_new_accumulator_instance,
+            mut main_accumulation_proof,
+            _config_phantom,
+            _predicate_phantom,
+        } = self;
+
+        // Ensure that prior data exist together.
+        assert_eq!(
+            main_input_instances.is_some(),
+            main_old_accumulator_instances.is_some()
+        );
+
+        // Ensure that prior data has the correct length.
+        assert!(
+            main_input_instances.is_none()
+                || main_input_instances.as_ref().unwrap().len() == P::PRIOR_MSG_LEN
+        );
+
+        assert!(
+            main_old_accumulator_instances.is_none()
+                || main_old_accumulator_instances.as_ref().unwrap().len() == P::PRIOR_MSG_LEN
+        );
+
+        /*
+         * Process the inputs
+         */
+
+        let base_case = main_input_instances.is_none();
+        let main_circuit_input_len = MainCircuit::<E, PC, P>::public_input_size();
+        let (main_input_instances, main_old_accumulator_instances) = if base_case {
+            (
+                vec![InputInstance::zero(main_circuit_input_len, MAKE_ZK); P::PRIOR_MSG_LEN],
```

Comment:
Maybe we should make this also `placeholder`, instead of `zero`?

Reply:
I have opened a PR in accumulation to apply this change (https://github.com/arkworks-rs/accumulation/pull/34).